choreolib/src/main/java/choreo/auto/AutoChooser.java

```diff
@@ -29,9 +32,17 @@
  */
 public class AutoChooser {
   /** A function that generates an {@link AutoRoutine} from an {@link AutoFactory}. */
-  public static interface AutoRoutineGenerator extends Function<AutoFactory, AutoRoutine> {
+  public interface AutoRoutineGenerator extends Function<AutoFactory, AutoRoutine> {
+    AutoFactory VOID_FACTORY = new AutoFactory(
+      Pose2d::new,
+      (pose2d, sampleType) -> {},
+      () -> false,
+      new Subsystem() {},
+      new AutoFactory.AutoBindings(),
+      Optional.empty()
+    );
     /** A generator that returns an auto routine that does nothing */
-    static final AutoRoutineGenerator NONE = factory -> AutoFactory.VOID_ROUTINE;
```

Comment:
Why change the way this is done?

Reply:
voidRoutine has to be an instance method because AutoRoutine now depends on the AutoFactory.